File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.*;

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for(int i=0; i<n; i++){
            answer[i] = Integer.toBinaryString(arr1[i] | arr2[i]);      // 두 수에 비트 연산자 or 적용 후(하나라도 1이면 1) 이진수 string으로 변환
            
            // 이진수 자릿수 맞춰주기 0을 n-answer[i].length()만큼 앞에 붙여줌
            // ex) 22 -> 10110, 14 -> 1110d이므로 비트연산을 적용하면 11110이 되는데 n은 여섯자리여야 하므로 앞에 0을 붙여서 01110이 됨
            answer[i] = "0".repeat(n-answer[i].length()) + answer[i];
            
            answer[i] = answer[i].replaceAll("1", "#");
            answer[i] = answer[i].replaceAll("0", " ");
        }
        
        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []

    for i in range(len(arr1)):
        num = arr1[i]|arr2[i]
        result = ''

        for bit in bin(num)[2:]:
            if bit == '1':
                result += '#'
            else:
                result += ' '
        if len(result) < n:
            result = ' '*(n-len(result))+result
        answer.append(result)
            
    return answer